perses/rjmc/geometry.py

```diff
@@ -291,8 +291,9 @@ def _logp_propose(self, top_proposal, old_positions, beta, new_positions=None, d
                 logZ_r = np.log((np.sqrt(2*np.pi)*(sigma_r/units.angstroms))) # CHECK DOMAIN AND UNITS
                 logp_r = self._bond_logq(r, bond, beta) - logZ_r
             else:
-                constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
-                r = constraint #set bond length to exactly constraint
+                if direction == 'forward':
+                    constraint = self._get_bond_constraint(atom, bond_atom, top_proposal.new_system)
```

Comment:
I'm mostly worried about bugs slipping through in case a bond as well as constraint are absent. The system might not work, but you could catch it earlier or later depending on where you have proper checks for it.